gossip/client_test.go

```diff
@@ -138,26 +139,34 @@ func TestClientGossip(t *testing.T) {
 		}
 	}()
 
-	if err := local.AddInfo("local-key", nil, time.Second); err != nil {
+	if err := local.AddInfo("local-key", nil, time.Hour); err != nil {
 		t.Fatal(err)
 	}
-	if err := remote.AddInfo("remote-key", nil, time.Second); err != nil {
+	if err := remote.AddInfo("remote-key", nil, time.Hour); err != nil {
 		t.Fatal(err)
 	}
 
 	// Use an insecure context. We're talking to tcp socket which are not in the certs.
 	rpcContext := rpc.NewContext(&base.Context{Insecure: true}, nil, stopper)
-	client.start(local, disconnected, rpcContext, stopper)
+	disconnected <- client
 
-	util.SucceedsSoon(t, func() error {
+	for ; true; time.Sleep(10 * time.Millisecond) {
+		select {
+		case <-disconnected:
+			// If the client wasn't able to connect, restart it.
```

Comment:
Yes, this is part of the flakiness fix. The default 3 sec connection dial timeout sometimes fires. If we don't start the client again the test will hang.